pkg/awsclient/sts_assume_role.go

```diff
@@ -0,0 +1,153 @@
+package awsclient
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-logr/logr"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/api/v1alpha1"
+	"github.com/openshift/aws-account-operator/config"
+	"github.com/rkt/rkt/tests/testutils/logger"
+	"regexp"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+var (
+	defaultSleepDelay = 500 * time.Millisecond
+)
+
+const (
+	controllerName = "account"
+)
+
+func matchSubstring(roleID, role string) (bool, error) {
+	matched, err := regexp.MatchString(roleID, role)
+	return matched, err
+}
+
+// getSTSCredentials returns STS credentials for the specified account ARN
+func GetSTSCredentials(
+	reqLogger logr.Logger,
+	client Client,
+	roleArn string,
+	externalID string,
+	roleSessionName string) (*sts.AssumeRoleOutput, error) {
+	// Default duration in seconds of the session token 3600. We need to have the roles policy
+	// changed if we want it to be longer than 3600 seconds
+	var roleSessionDuration int64 = 3600
+	reqLogger.Info(fmt.Sprintf("Creating STS credentials for AWS ARN: %s", roleArn))
+	// Build input for AssumeRole
+	assumeRoleInput := sts.AssumeRoleInput{
+		DurationSeconds: &roleSessionDuration,
+		RoleArn:         &roleArn,
+		RoleSessionName: &roleSessionName,
+	}
+	if externalID != "" {
```

Comment:
Thinking out loud here:
I'm wondering if there's even a difference in the aws api between 
```
sts.AssumeRoleInput{
  ExternalId: nil
}
```
and 
```
sts.AssumeRoleInput{
  ExternalId: aws.String("")
}
```
. Cause for string itself, there's no difference between a nil string and "". If there were no difference in the api between nil and a pointer to `""`, then we could just get rid of this check and always just pass the parameter to the AssumeRoleInput.